mechanics/tools/gdocs-migrate/forest.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+type File struct {
+	id       string
+	name     string
+	parentID string
+	isFolder bool
+}
+
+type Forest struct {
+	fileByID map[string]File
+	children map[string][]File
+	leaves   []File
+}
+
+func NewForest() Forest {
+	return Forest{
+		fileByID: make(map[string]File),
+		children: make(map[string][]File),
+	}
+}
+
+func (fo *Forest) Add(f File) {
+	fo.fileByID[f.id] = f
+
+	fo.children[f.parentID] = append(fo.children[f.parentID], f)
+	if !f.isFolder {
```

Comment:
I think this is fine; another way to do this would be to put files into either `fileByID` or `children` based on whether or not they are a folder. It would probably require a slightly more complex structure (`struct {File, [] File}`) instead of the current slice.

Reply:
I'll keep it this way for now and see where this leads us, and will change the interface/data as needed.